File: ExampleApp/app/src/main/java/com/motiv/example/AuthToken.java
package com.motiv.example;

import android.widget.ImageView;
import androidx.annotation.NonNull;
import androidx.databinding.BindingAdapter;
import androidx.fragment.app.*;
import androidx.room.*;
import com.bumptech.glide.Glide;
import com.google.gson.*;
import com.google.gson.annotations.*;
import com.google.gson.reflect.*;
import dagger.*;
import dagger.android.*;
import dagger.android.support.*;
import java.util.*;
import java.util.concurrent.*;
import javax.inject.*;

@Entity(tableName = "authtoken")
public class AuthToken {

    private static final Gson gson = new Gson();
    @NonNull @PrimaryKey private java.lang.String id = UUID.randomUUID().toString();

    @ColumnInfo(name = "token")
    @SerializedName("token")
    private java.lang.String token;

    public java.lang.String getId() {

        return this.id;
    }

    public void setId(java.lang.String id) {
        this.id = id;
    }

    public java.lang.String getToken() {

        return this.token;
    }

    public void setToken(java.lang.String token) {
        this.token = token;
    }

    @BindingAdapter({"bind:imageUrl"})
    public static void loadImage(ImageView view, java.lang.String url) {
        Glide.with(view.getContext()).load(url).into(view);
    }

    public static AuthToken fromJson(String json) {
        return gson.fromJson(json, AuthToken.class);
    }

    public static String toJson(AuthToken object) {
        return gson.toJson(object, AuthToken.class);
    }

    public static String toJsonArray(AuthToken[] array) {
        return gson.toJson(array, AuthToken[].class);
    }

    public static AuthToken[] fromJsonArray(String array) {
        return gson.fromJson(array, AuthToken[].class);
    }
}

File: ExampleApp/app/src/main/java/com/motiv/example/AuthToken.kt
 
package com.motiv.example
import androidx.annotation.NonNull
import com.google.gson.*
import com.google.gson.annotations.*
import com.google.gson.reflect.*
import io.realm.*
import java.util.*
import java.util.concurrent.*

public open class AuthToken : RealmObject() {

    @NonNull private var id: String = UUID.randomUUID().toString()

    @SerializedName("token")

    private var token: String = ""

    fun getId(): String {
        return this.id
    } fun setId(id: String) {
        this.id = id
    } fun getToken(): String {
        return this.token
    } fun setToken(token: String) {
        this.token = token
    } companion object {
        val gson: Gson = Gson()
        fun fromJson(json: String): AuthToken {
            return gson.fromJson(json, AuthToken::class.java)
        }

        fun toJson(obj: AuthToken): String {
            return gson.toJson(obj, AuthToken::class.java)
        }

        fun toJsonArray(obj: AuthToken): String {
            return gson.toJson(obj, Array<AuthToken>::class.java)
        }

        fun fromJsonArray(json: String): Array<AuthToken> {
            return gson.fromJson(json, Array<AuthToken>::class.java)
        }
    }
}
